main.rb

```diff
@@ -0,0 +1,185 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1, true, 'hi', []].my_all? != [1, true, 'hi', []].all?
[1, false, 'hi', []].my_all? != [1, false, 'hi', []].all?
[1,2,3,4,5].my_all?(Integer) != [1,2,3,4,5].all?(Integer)
["word",2,3,4,5].my_all?(Integer) != ["word",2,3,4,5].all?(Integer)
['saheed', 'oladele', 'suretrust'].my_all?(/d/) != ['saheed', 'oladele', 'suretrust'].all?(/d/)
[1,2,3,4,5].my_all?(3) != [1,2,3,4,5].all?(3)
```